diff --git a/src/ngx_http_memc_util.c b/src/ngx_http_memc_util.c
index 090bf43..bb8476a 100644
--- a/src/ngx_http_memc_util.c
+++ b/src/ngx_http_memc_util.c
@@ -141,12 +141,14 @@ ngx_http_memc_upstream_add(ngx_http_request_t *r, ngx_url_t *url)
             continue;
         }
 
+#if (nginx_version < 1011006)
         if (uscfp[i]->default_port && url->default_port
             && uscfp[i]->default_port != url->default_port)
         {
             dd("upstream_add: default_port not match");
             continue;
         }
+#endif
 
         return uscfp[i];
     }
